packages/perpswap/src/contracts/vault.rs

```diff
@@ -19,12 +19,45 @@ pub struct InstantiateMsg {
     pub markets_allocation_bps: HashMap<String, u16>,
 }
 
+/// Denomination of USDC token
+#[derive(serde::Serialize, serde::Deserialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum UsdcDenom {
+    /// CW20 USDC
+    CW20(Addr),
+    /// IBC USDC
+    IBC(String),
+    /// Native USDC
+    Native(String),
+}
+impl fmt::Display for UsdcDenom {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            UsdcDenom::CW20(addr) => write!(f, "{}", addr),
+            UsdcDenom::IBC(path) => write!(f, "{}", path),
+            UsdcDenom::Native(denom) => write!(f, "{}", denom),
+        }
+    }
+}
+
+impl From<String> for UsdcDenom {
+    fn from(s: String) -> Self {
+        if s.starts_with("osmo1") {
```

Comment:
This isn't something we should do in general, since this contract needs to work on multiple chains. Instead, if there's user input for providing an asset, it should receive an enum allowing the caller to specify whether it's CW20 or native.